Named tuples pretty format

Sasao4o opened the pull request:
Example:
 

```sql
-- Query
SELECT
    i.InstanceType,
    i.VCpuInfo.1 AS cpu,
    i.InstanceStorageInfo
FROM
(
    SELECT arrayJoin(InstanceTypes) AS i
    FROM instances
)
WHERE i.InstanceStorageSupported
ORDER BY cpu DESC
LIMIT 10


-- Output
┌─i.InstanceType─┬─cpu─┬─i.InstanceStorageInfo───────────────────┐
│ test.large     │  64 │ {                                      ↴│
│                │     │↳  "Disks": [                           ↴│
│                │     │↳   {                                   ↴│
│                │     │↳    "Count": 2,                        ↴│
│                │     │↳    "SizeGB": 1900,                    ↴│
│                │     │↳    "Type": "ssd"                      ↴│
│                │     │↳   }                                   ↴│
│                │     │↳  ],                                   ↴│
│                │     │↳  "EbsOptimizedSupport": "required",   ↴│
│                │     │↳  "EncryptionSupport": "required",     ↴│
│                │     │↳  "TotalSizeGB": 3800                  ↴│
│                │     │↳ }                                      │
│ test.xlarge    │  32 │ {                                      ↴│
│                │     │↳  "Disks": [                           ↴│
│                │     │↳   {                                   ↴│
│                │     │↳    "Count": 4,                        ↴│
│                │     │↳    "SizeGB": 2000,                    ↴│
│                │     │↳    "Type": "ssd"                      ↴│
│                │     │↳   }                                   ↴│
│                │     │↳  ],                                   ↴│
│                │     │↳  "EbsOptimizedSupport": "unsupported",↴│
│                │     │↳  "EncryptionSupport": "unsupported",  ↴│
│                │     │↳  "TotalSizeGB": 8000                  ↴│
│                │     │↳ }                                      │
└────────────────┴─────┴─────────────────────────────────────────┘
 ```

### Changelog category (leave one):
- Improvement


### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):

In Pretty format, named tuples are now displayed as Pretty JSON. This closes #65022

Sasao4o committed:
Improve Pretty format: display named tuples as JSON

Sasao4o committed:
Add Test for NamedTuple in PrettyFormat

pamarcos commented on src/DataTypes/Serializations/SerializationTuple.cpp:
```diff
@@ -153,6 +156,7 @@ void SerializationTuple::serializeText(const IColumn & column, size_t row_num, W
         elems[i]->serializeTextQuoted(extractElementColumn(column, i), row_num, ostr, settings);
     }
     writeChar(')', ostr);
+    }
```
Please indent this properly

pamarcos commented on src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:
```diff
@@ -32,6 +32,9 @@ PrettyBlockOutputFormat::PrettyBlockOutputFormat(
     }
+    format_settings.is_pretty_format = true;
+    format_settings.json = FormatSettings::JSON{};
+    format_settings.json.pretty_print_indent_multiplier = 1;
```
Why hardcoding this here instead of using the provided value?

Sasao4o replied:
I thought JSON settings shouldn't affect the pretty format, also the default value of the print_indent_multiplier is 4, which is not suitable (it's also hardcoded in JSONRowOutputFormat.cpp), so I reset the json settings and changed the multiplier

Sasao4o committed:
Style Fix

Sasao4o committed:
Rename is_pretty_format to pretty_format

Sasao4o committed:
Merge remote-tracking branch 'upstream/master' into named-tuples-pretty-format

Sasao4o committed:
Add New FormatSetting output_format_pretty_named_tuples_as_json

Sasao4o committed:
Add backward compatibility for failed tests

Sasao4o committed:
Fix Typo + Remove Whitespace

Sasao4o committed:
Add OrderBy To Ensure Determnistic test output

Sasao4o committed:
Improve Test (Remove UNION ALL)

pamarcos committed:
Use PrettyMonoBlock to make sure rows are always glued together

pamarcos committed:
Merge branch 'master' into named-tuples-pretty-format

pamarcos committed:
Merge branch 'master' into named-tuples-pretty-format

pamarcos committed:
Fix SettingsChangesHistory.cpp style

Remove trailing whitespaces added by GitHub web UI when solving the merge conflict
